src/routing/PrivateRoute.tsx

```diff
@@ -1,14 +1,30 @@
 import React from 'react';
 
-import { Redirect, Route } from 'react-router-dom';
+import { Redirect, Route, RouteComponentProps } from 'react-router-dom';
 
 import { ROUTE_PATHS } from '@config';
 import { useAccounts } from '@services/Store';
+import { translateRaw } from '@translations';
+import { IAppRoute } from '@types';
+import { useEffectOnce } from '@vendor';
 
-interface PrivateRouteProps {
-  [key: string]: any;
+interface PrivateRouteProps extends IAppRoute {
+  key: number;
 }
 
+interface PageTitleProps extends RouteComponentProps {
+  pageComponent: any;
+  title?: string;
+}
+
+const PageTitleRoute = ({ pageComponent: Component, title, ...props }: PageTitleProps) => {
+  useEffectOnce(() => {
+    document.title =
+      translateRaw('PAGE_TITLE_PREPEND') + title || translateRaw('DEFAULT_PAGE_TITLE');
```

Comment:
Gives: "MyCrypto - undefined" if the title is undefined.

Reply:
nice catch :)